homeassistant/components/synology_dsm/__init__.py

```diff
@@ -562,6 +631,61 @@ async def async_added_to_hass(self):
         self.async_on_remove(self._api.subscribe(self._api_key, self.unique_id))
 
 
+class SynologyDSMCoordinatorEntity(CoordinatorEntity):
```

Comment:
It looks like we now have 3 entity classes. Add `SynologyDisptacherDSMEntity` and move `should_poll`, `async_update`, `async_added_to_hass` to it, then make `SynologyDSMEntity` a base class of it and `SynologyDSMCoordinatorEntity`.  Remove the duplication from `SynologyDSMCoordinatorEntity`

Reply:
How is about to keeping `class SynologyDSMEntity(Entity)` as is and add `SynologyDSMEntity` as additional baseclass to `SynologyDSMCoordinatorEntity` and remove duplicates from `SynologyDSMCoordinatorEntity` (_which are already defined in `SynologyDSMEntity`_).
To keep correct MRO I think `SynologyDSMCoordinatorEntity(CoordinatorEntity, SynologyDSMEntity)` should be correct 🤔

Reply:
ok, now there are following entity classes:

`SynologyDSMBaseEntity`:
containing all synology relevant things

`SynologyDSMDispatcherEntity(SynologyDSMBaseEntity, Entity)`:
contain `should_poll` , `async_update` and `async_added_to_hass`

`SynologyDSMCoordinatorEntity(SynologyDSMBaseEntity, CoordinatorEntity)`:
implements `CoordinatorEntity`

`SynologyDSMDeviceEntity(SynologyDSMDispatcherEntity)`:
only base is changed to `SynologyDSMDispatcherEntity`

Reply:
I'll give this a test on my production instances

Reply:
```
2021-01-31 18:08:31 DEBUG (MainThread) [homeassistant.runner] Calling executor with function: <bound method SynologyDSM.update of <synology_dsm.synology_dsm.SynologyDSM object at 0x7f02d8410ca0>>, args: (True,), kwargs: {}
2021-01-31 18:08:32 DEBUG (MainThread) [homeassistant.runner] Calling executor with function: <bound method SynoSurveillanceStation.get_home_mode_status of <synology_dsm.api.surveillance_station.SynoSurveillanceStation object at 0x7f02d7b62370>>, args: (), kwargs: {}
2021-01-31 18:08:35 DEBUG (MainThread) [homeassistant.runner] Calling executor with function: <bound method SynoSurveillanceStation.update of <synology_dsm.api.surveillance_station.SynoSurveillanceStation object at 0x7f02d7b62370>>, args: (), kwargs: {}
```

3 executor jumps.  Will verify they are all needed

Reply:
Wow this make a lot of api calls under the hood to update the camera.  
https://github.com/hacf-fr/synologydsm-api/blob/master/src/synology_dsm/api/surveillance_station/__init__.py#L25

Reply:
LGTM 👍

Reply:
If can wait some minuts, than my build host will have created docker images based on this last commit ;)